ipfsconn/ipfshttp/config.go

```diff
@@ -137,22 +154,40 @@ func (cfg *Config) LoadJSON(raw []byte) error {
 	cfg.ProxyAddr = proxyAddr
 	cfg.NodeAddr = nodeAddr
 
-	// errors ignored as Validate() below will catch them
-	t, _ := time.ParseDuration(jcfg.ProxyReadTimeout)
+	// only overwrite defaults when we can parse the time
+	// Note for these 0 is a valid value.
+	t, err := time.ParseDuration(jcfg.ProxyReadTimeout)
+	if err != nil {
+		return fmt.Errorf("error parsing proxy_read_timeout: %s", err)
+	}
```

Comment:
Because now 0 is a valid value, and it's the default value returned by ParseDuration in case of error.

So I need to check the error and abort of the user wrote something like `-123s` or `abcd` in a time duration field.

Reply:
Before, since we didn't consider 0 valid, we could simplify and ignore parsing errors since it all accounted to the same thing.